### PR TITLE
Implemented top ram processes feature from #81

### PR DESCRIPTION
**Pull request by jevonmao**
Sorry, for some reason this branch includes my previous commits that connect CPU top process refresh rate to SMC refresh setting. That is yet to be approved. But I implemented top ram process feature, it's all working, I just think you will prefer if I refactored the code into a seperate view model.

### CODE REVIEW

**Comment by gao-sun** [comment]
thanks, will try to take a look tonight and give you my feedback

**Comment by jevonmao** [comment]
Thanks. Just keep in mind that this is still WIP.

I specifically need help with updating the ram information redraw the views. Because right now they won't update at all - I noticed something called refreshable protocol and I think it is a custom view updating thing. I'm not sure why is that needed? and how to update the SwiftUI view?

**Comment by gao-sun** [comment]
would you mind to fix the swiftformat issues?

**Comment by gao-sun** [comment]
taking a look tonight

File: eul/Components/RamProcessRowView.swift
Comment (gao-sun) [inline eul/Components/RamProcessRowView.swift:57]: maybe this can be wrapped into a `View`?

File: eul/Store/MemoryTopStore.swift
Comment (gao-sun) [inline eul/Store/MemoryTopStore.swift:58]: maybe we can re-use the top command for better performance. consider merging CPU/mem top stores into one store?

**Review by gao-sun — COMMENTED** [review]
overall looks great. let me know if you have any questions

**Comment by jevonmao** [comment]
@gao-sun Regarding your point about merging CPU/mem command, I think it would work. Because for CPU it is ordered by CPU usage to get top processes, but for ram it is ordered by ram usage size. We would, however, merge cpu and ram top stores into 1 store.

**Comment by gao-sun** [comment]
@jevonmao oh shoot, forgot the difference of ordering. I'm good with 2 stores then! I think you can go with a shared row parsing util for `top` result instead of merging them into one store. sorry about my thoughtlessness

**Comment by jevonmao** [comment]
@gao-sun How about 1 store with 2 different published arrays for holding ram and CPU? And still partially share some parsing code between the 2.

**Comment by gao-sun** [comment]
@jevonmao yes. sounds great!

**Comment by jevonmao** [comment]
Everything is finished at last. Feel free to review and merge @gao-sun !

File: eul.xcodeproj/project.pbxproj
Comment (gao-sun) [inline eul.xcodeproj/project.pbxproj:1549]: mind revert the team?

File: eul/Store/TopStore.swift
Comment (gao-sun) [inline eul/Store/TopStore.swift:22]:
```suggestion
    @Published var cpuTopProcesses: [ProcessCpuUsage] = []
```

File: eul/Components/RamProcessRowView.swift
Comment (gao-sun) [inline eul/Components/RamProcessRowView.swift:12]: I think you can re-use `ProcessRowView` and use `valueViewBuilder` for customized value display - would you mind to take a look?

**Review by gao-sun — COMMENTED** [review]
I think the direction is correct, just few more comments. Will be good to merge if they are cleared.

**Comment by jevonmao** [comment]
I fixed the things you commented

**Review by gao-sun — DISMISSED** [review]
thanks! looks great. mind me to do some minor modification before merging?

**Comment by jevonmao** [comment]
Yeah sure go ahead

**Comment by gao-sun** [comment]
looks like I cannot update your branch directly. would you mind to resolve conflicts and remove `RamProcessComponentView.swift`(looks like it's not in use anymore)? then the PR can be merged

**Comment by gao-sun** [comment]
btw, i can add you into repo collaborators if you are interested. so you can develop using this repo directly

**Comment by jevonmao** [comment]
Yes please. Thank you!

**Comment by jevonmao** [comment]
@gao-sun Removed the unused file and merged conflicts. Interesting that I am not allowed to merge this branch despite that I have write access to this repository.

**Comment by gao-sun** [comment]
because i checked "dismiss stale approval", which means if the branch has been updated, the pr will need a new approval. just took a quick look and it looks good. i'll follow up tonight (approve/merge/bump version), just sit back and wait for the launch. cheers!

**Review by gao-sun — APPROVED** [review]
I think there's no major issues. approve to unblock. thank you for contributing!

**Comment by gao-sun** [comment]
have to handle some stuff tonight, merged first. stay tuned on the new release

**Comment by jevonmao** [comment]
@gao-sun I see you did some polishing and refactor with top ram related code, great job. What do you mean by `TO-DO: extract store logic` though?

**Comment by gao-sun** [comment]
@jevonmao I saw a familiar pattern is presented in top store so it'll be great to extract it for cleaner code

**Pull request merged**
